cleanlab/count.py

```diff
@@ -96,18 +85,45 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
       The `confident_joint` can be computed using :py:func:`count.compute_confident_joint <cleanlab.count.compute_confident_joint>`.
       If not provided, it is computed from the given (noisy) `labels` and `pred_probs`.
 
+    estimation_method :
+      Method for estimating the number of label issues in dataset by counting the examples in the off-diagonal of the `confident_joint` ``P(label=i, true_label=j)``.
+       - ``'off_diagonal'``: Counts the number of examples in the off-diagonal of the `confident_joint`. Returns the same value as ``sum(find_label_issues(filter_by='confident_learning'))``
+       - ``'off_diagonal_recalibrated'``: Calibrates confident joint estimate ``P(label=i, true_label=j)`` such that
+       ``np.sum(cj) == len(labels)`` and ``np.sum(cj, axis = 1) == np.bincount(labels)`` before counting the number
+       of examples in the off-diagonal. Number will always be equal to or greater than
+       ``estimate_issues='off_diagonal'``. You can use this value as the cutoff threshold used with ranking/scoring
+       functions from :py:mod:`cleanlab.rank` with `num_label_issues` over ``estimation_method='off_diagonal'`` in
+       two cases:
+          1. As we add more label and data quality scoring functions in :py:mod:`cleanlab.rank`, this approach will always work.
+          2. If you have a custom score to rank your data by label quality and you just need to know the cut-off of likely label issues.
+
     Returns
     -------
     num_issues : int
       The estimated number of examples with label issues in the dataset.
     """
+    valid_methods = ["off_diagonal", "off_diagonal_recalibrated"]
 
     if confident_joint is None:
-        confident_joint = compute_confident_joint(labels=labels, pred_probs=pred_probs)
-    # Normalize confident joint so that it estimates the joint, p(labels,y)
-    joint = confident_joint / float(np.sum(confident_joint))
-    frac_issues = 1.0 - joint.trace()
-    num_issues = np.rint(frac_issues * len(labels)).astype(int)
+        confident_joint = compute_confident_joint(
+            labels=labels, pred_probs=pred_probs, calibrate=False
+        )
+
+    if estimation_method is "off_diagonal":
+        num_issues = np.sum(confident_joint) - np.trace(confident_joint)
+    elif estimation_method is "off_diagonal_recalibrated":
+        # Normalize confident joint so that it estimates the joint, p(labels,y)
+        confident_joint = calibrate_confident_joint(confident_joint, labels)
+        joint = confident_joint / float(np.sum(confident_joint))
```

Comment:
```suggestion
        joint = estimate_joint(labels, pred_probs, confident_joint=confident_joint)
```

this is the cleanlab-way to do it in one line of code. achieves identical result as the previous code just uses the actual function (which it should)

Reply:
Note that `confident_joint` is not calibrated but this method calibrates it and then normalizes (makes it sum to 1) which is what the line above did.

you could add a comment above this line that says:

`# estimate_joint calibrates the row sums to match the prior distribution of given labels and normalizes to sum to 1`